### PR TITLE
Improve types for tuples

### PR DESCRIPTION
**Pull request by krzkaczor**
Closes: https://github.com/ethereum-ts/TypeChain/issues/232

### CODE REVIEW

File: .changeset/fair-paws-pay.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+x(overrides?: CallOverrides): Promise<{0: BigNumber}>;
+```
+
+New output is:
+
+```typescript
+x(overrides?: CallOverrides): Promise<[BigNumber]>;
+```
+
+The difference is that now you can standard array destruction while working with output types.
```
Comment (quezak) [inline .changeset/fair-paws-pay.md:27]: image
([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment))
```suggestion
The difference is that now you can use standard array destructuring while working with output types.
There should be no changes needed to existing code.
```

File: packages/target-ethers-v5/src/codegen/types.ts [hunk 2]
```diff
@@ -82,11 +79,15 @@ export function generateTupleType(tuple: TupleType, generator: (evmType: EvmType
...
 
-export function generateOutputTupleType(tuple: TupleType) {
-  return (
-    '{' +
-    tuple.components.map((component) => `${component.name}: ${generateOutputType(component.type)} ,`).join('\n') +
-    tuple.components.map((component, index) => `${index}: ${generateOutputType(component.type)}`).join(', ') +
-    '}'
-  )
+/**
+ * always return an array type if there are named outputs, attach them to that type
```
Comment (quezak) [inline packages/target-ethers-v5/src/codegen/types.ts:83]:
```suggestion
 * always return an array type; if there are named outputs, merge them to that type
```

File: packages/target-truffle-v5-test/types/truffle-contracts/DataTypesInput.d.ts [hunk 3]
```diff
@@ -67,7 +67,7 @@ export interface DataTypesInputInstance extends Truffle.ContractInstance {
     input1: number | BN | string,
     input2: number | BN | string,
     txDetails?: Truffle.TransactionDetails
-  ): Promise<[BN, BN]>;
+  ): Promise<{ 0: BN; 1: BN }>;
```
Comment (quezak) [inline packages/target-truffle-v5-test/types/truffle-contracts/DataTypesInput.d.ts:70]: funny that for truffle the change is excatly the opposite of the ethers change :sweat_smile:

**Review by quezak — APPROVED** [review]

File: packages/target-truffle-v5-test/types/truffle-contracts/DataTypesInput.d.ts (hunk 3 above)
Comment (krzkaczor) [reply packages/target-truffle-v5-test/types/truffle-contracts/DataTypesInput.d.ts:70]: Yup :D but anyway, now types correspond to real runtime values so it's fine.

**Pull request merged**